flex-templates/java/regional_dlp_transform/src/main/java/com/google/cloud/blueprints/datawarehouse/BigQueryDynamicWriteTransform.java

```diff
@@ -0,0 +1,150 @@
+import org.slf4j.LoggerFactory;
+
+@AutoValue
+@SuppressWarnings("serial")
+public abstract class BigQueryDynamicWriteTransform
+    extends PTransform<PCollection<KV<String, TableRow>>, WriteResult> {
+  public static final Logger LOG = LoggerFactory.getLogger(BigQueryDynamicWriteTransform.class);
+
+  public abstract String projectId();
+
+  public abstract String datasetId();
+
+  public static Builder newBuilder() {
+    return new AutoValue_BigQueryDynamicWriteTransform.Builder();
+  }
+
+  private static final String jsonSchema = "{"
+      + "\"fields\":["
+      + "{ \"description\": \"Card_Type_Code\", \"mode\": \"REQUIRED\", \"name\": \"Card_Type_Code\", \"type\": \"STRING\"},"
+      + "{ \"description\": \"Card_Type_Full_Name\", \"mode\": \"REQUIRED\", \"name\": \"Card_Type_Full_Name\", \"type\": \"STRING\"},"
+      + "{ \"description\": \"Issuing_Bank\", \"mode\": \"REQUIRED\", \"name\": \"Issuing_Bank\", \"type\": \"STRING\" },"
+      + "{ \"description\": \"Card_Number\", \"mode\": \"REQUIRED\", \"name\": \"Card_Number\", \"type\": \"STRING\" },"
+      + "{ \"description\": \"Card_Holders_Name\", \"mode\": \"REQUIRED\", \"name\": \"Card_Holders_Name\", \"type\": \"STRING\"},"
+      + "{ \"description\": \"CVV2\", \"mode\": \"REQUIRED\", \"name\": \"CVVCVV2\", \"type\": \"STRING\"},"
+      + "{ \"description\": \"Issue_Date\", \"mode\": \"REQUIRED\", \"name\": \"Issue_Date\", \"type\": \"STRING\"},"
+      + "{ \"description\": \"Expiry_Date\", \"mode\": \"REQUIRED\", \"name\": \"Expiry_Date\", \"type\": \"STRING\"},"
+      + "{ \"description\": \"Billing_Date\", \"mode\": \"REQUIRED\", \"name\": \"Billing_Date\", \"type\": \"STRING\"},"
+      + "{ \"description\": \"Card_PIN\", \"mode\": \"REQUIRED\", \"name\": \"Card_PIN\", \"type\": \"STRING\"},"
+      + "{ \"description\": \"Credit_Limit\", \"mode\": \"REQUIRED\", \"name\": \"Credit_Limit\", \"type\": \"STRING\"}"
+      + "]"
+      + "}";
+
+  @AutoValue.Builder
+  public abstract static class Builder {
+    public abstract Builder setDatasetId(String projectId);
+
+    public abstract Builder setProjectId(String datasetId);
+
+    public abstract BigQueryDynamicWriteTransform build();
+  }
+
+  @Override
+  public WriteResult expand(PCollection<KV<String, TableRow>> input) {
+
+    return input.apply(
+        "BQ Write",
+        BigQueryIO.<KV<String, TableRow>>write()
+            // .to(new BQDestination(datasetId(), projectId()))
+            .to(projectId() + ":" + datasetId() + "." + "cc_10000_records")
+            .withFormatFunction(
+                element -> {
+                  return element.getValue();
+                })
+            .withWriteDisposition(BigQueryIO.Write.WriteDisposition.WRITE_APPEND)
+            .withCreateDisposition(BigQueryIO.Write.CreateDisposition.CREATE_IF_NEEDED)
+            .withoutValidation()
+            .withMethod(BigQueryIO.Write.Method.FILE_LOADS)
+            .withTriggeringFrequency(Duration.standardSeconds(10))
```

Comment:
I see a magic number of 10.  plz avoid those.  use a const and add a comment with the const indicating why 10 seconds is important.  if it was randomly selected, let's think of a better reason/reference to use.  this is especially important when ingesting 100TB of data